src/components/Pagination/index.tsx

```diff
@@ -0,0 +1,99 @@
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  ChevronsLeftIcon,
+  ChevronsRightIcon,
+} from 'lucide-react';
+
+import useSetSearchParams from '@/hooks/useSetSearchParam';
+
+import * as S from './Pagination.css';
+
+interface PaginationProps {
+  totalPage: number;
+  currentPage: number;
+}
+
+const NUM = 5;
+
+function Pagination({ pageInfo }: { pageInfo: PaginationProps }) {
+  const { totalPage, currentPage } = pageInfo;
+  const { setSingleSearchParam } = useSetSearchParams();
+
+  const handlePageChange = (newPage: number) => {
+    setSingleSearchParam('page', newPage.toString());
+  };
+
+  const half = Math.floor(NUM / 2);
+  const startPage = Math.max(1, Math.min(currentPage + 1 - half, totalPage - NUM + 1));
+  const endPage = Math.min(totalPage, startPage + NUM - 1);
+
+  const numberStyle = (page: number) => {
+    return currentPage + 1 === page ? S.base.currentNumbers : S.base.numbers;
+  };
+
+  const chevronLeftStyle = () => {
+    return currentPage + 1 === 1 ? S.base.numbers : S.base.currentNumbers;
+  };
+
+  const chevronRightStyle = () => {
+    return currentPage + 1 === totalPage ? S.base.numbers : S.base.currentNumbers;
+  };
+
+  return (
+    <div className={S.pagination}>
+      <div className={S.chevronButton}>
+        <button
+          className={chevronLeftStyle()}
+          type="button"
+          onClick={() => handlePageChange(1)}
+          disabled={currentPage + 1 === 1}
```

Comment:
```suggestion
const LASTPAGE = 1
----
          disabled={currentPage + 1 === LASTPAGE}
```
어디서는 또 다른 모종의 이슈로 LastPage가 다를 수 있으니 1이라는 값보다는 상수로 값의 특징을 명확하게 알려줘도 좋을 것 같아요

Reply:
LAST_PAGE는 유동적이여서 일단 FIRST_PAGE만 반영했습니다!